### PR TITLE
(MODULES-9479) Fix nested array merge behavior

### PR DESCRIPTION
**Pull request by seanmil**
Fix regression in behavior introduced in the 5.2.0 release
with the fix for MODULES-8287. Restore the behavior where arrays
nested inside hashes will be merged.

### CODE REVIEW

File: spec/acceptance/force_spec.rb [hunk 1]
```diff
@@ -91,6 +91,33 @@
...
+          target  => '#{@basedir}/file',
+          content => '{"one": [2]}',
+        }
+      MANIFEST
+    end
+
+    it 'applies the manifest twice with no stderr' do
+      idempotent_apply(pp)
+      expect(file("#{@basedir}/file")).to be_file
+      expect(file("#{@basedir}/file").content).to match '{"one":[1,2]}' || '{"one":[1,2]}\n'
```
Comment (tphoney) [inline spec/acceptance/force_spec.rb]: could it be ```expect(file("#{@basedir}/file").content).to match '{"one":[1,2]}\R?'``` ?

Comment (tphoney) [reply spec/acceptance/force_spec.rb]: https://github.com/puppetlabs/puppetlabs-concat/blob/master/spec/acceptance/newline_spec.rb#L51 ?

**Comment by tphoney** [comment]
apologies for closing the pr, it was to force kick appveyor

File: spec/acceptance/force_spec.rb (hunk 1 above)
Comment (MartyEwings) [reply spec/acceptance/force_spec.rb]: what about `.to contain '{"one":[1,2]}'` we dont specifically care if the new line is there or not,

Comment (tphoney) [reply spec/acceptance/force_spec.rb]: 👍

**Comment by MartyEwings** [comment]
[martin.ewings@martin puppetlabs-concat]$ bundle exec rake litmus:acceptance:parallel
┌ [✔] Running against 24 targets.
├──[✔] aubjs7eirul561d.delivery.puppetlabs.net, redhat-5-x86_64
├──[✔] i744sb977vbhqhx.delivery.puppetlabs.net, redhat-6-x86_64
├──[✔] b8qel0fam1uhs0i.delivery.puppetlabs.net, redhat-7-x86_64
├──[✔] vsmrt4cskliany3.delivery.puppetlabs.net, redhat-8-x86_64
├──[✔] bc76jeqm95sn2wi.delivery.puppetlabs.net, centos-5-x86_64
├──[✔] dnrwy9iqcch5epx.delivery.puppetlabs.net, centos-6-x86_64
├──[✔] aey3d9m79b96l8l.delivery.puppetlabs.net, centos-7-x86_64
├──[✔] sa44gtfwwn4jkhz.delivery.puppetlabs.net, oracle-5-x86_64
├──[✔] eajc86s641vk6fa.delivery.puppetlabs.net, oracle-6-x86_64
├──[✔] vmk1w2hr4owpuuw.delivery.puppetlabs.net, oracle-7-x86_64
├──[✔] of203acpk627l9h.delivery.puppetlabs.net, scientific-6-x86_64
├──[✔] cvyvvgjw8b3k4rp.delivery.puppetlabs.net, scientific-7-x86_64
├──[✔] oaonypaz7l4co4j.delivery.puppetlabs.net, debian-8-x86_64
├──[✔] mi1uam0y9iqlxx0.delivery.puppetlabs.net, debian-9-x86_64
├──[✔] u3xz6vudbn48cia.delivery.puppetlabs.net, sles-11-x86_64
├──[✔] yq28k3l8xcae63n.delivery.puppetlabs.net, sles-12-x86_64
├──[✔] bxx65yroo3zjs5b.delivery.puppetlabs.net, sles-15-x86_64
├──[✔] spgn3fg6m0a5mtd.delivery.puppetlabs.net, ubuntu-1404-x86_64
├──[✔] vffhk1fz53zb8qm.delivery.puppetlabs.net, ubuntu-1604-x86_64
├──[✔] tf3uvkv2zprobfb.delivery.puppetlabs.net, ubuntu-1804-x86_64
├──[✔] u241ctkdgv1fomf.delivery.puppetlabs.net, win-2008r2-x86_64
├──[✔] f0jko6c2dwemqhb.delivery.puppetlabs.net, win-2012r2-x86_64
├──[✔] ffvsdj9e71p60bc.delivery.puppetlabs.net, win-2016-x86_64
└──[✔] g1uzc39kgfejb82.delivery.puppetlabs.net, win-10-pro-x86_64
================

**Pull request merged**